src/main/java/io/getunleash/FeatureDefinition.java

```diff
@@ -0,0 +1,35 @@
+package io.getunleash;
+
+import io.getunleash.engine.FeatureDef;
+import java.util.Optional;
+
+public class FeatureDefinition {
+
+    private final String name;
+    private final Optional<String> type;
+    private final String project;
+
+    public FeatureDefinition(FeatureDef source) {
```

Comment:
Isn't FeatureDef and FeatureDefinition exactly the same? If so we could drop one of the two

Reply:
They contain the same data but FeatureDefinition is an Yggdrasil entity and I don't like leaking types from transient dependencies

Reply:
I was going to ask... is it? But I think you swap them as FeatureDef is an Yggdrasil entity. So yes, given that I think it's ok to have the two